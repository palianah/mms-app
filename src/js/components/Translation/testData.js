// @flow

window.app = {
  translations: {
    EN: {
      "App": {
        "Name": "Github Issues"
      }
    },
    DE: {
      "App": {
        "Name": "DE Github Issues"
      }
    },
  },
  languages: ['EN', 'DE'],
  curLang: 'EN',
  defLang: 'EN'
};
