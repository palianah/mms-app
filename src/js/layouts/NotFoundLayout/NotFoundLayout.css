/* Not Found Layout */
/* ---------------- */

.NotFoundLayout {
  width: 100%;
  height: 100%;
  margin: 0px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}