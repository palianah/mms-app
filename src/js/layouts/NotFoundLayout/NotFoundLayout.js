// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import InfoMsg  from '../../components/InfoMsg/InfoMsg';
import Translation, { text }  from '../../components/Translation/Translation';
import { ROUTE_HOME } from '../../constants/routes';
import { ICON_STOP } from '../../constants/icons';
import './NotFoundLayout.css';

type Props = {
  history: Object,
  location: Object,
  match: Object,
};


/**
* NotFound Layout
*/
const NotFoundLayout = (props: Props) => (
  <section className="NotFoundLayout">
    <InfoMsg icon={ICON_STOP} msg={text('Message', 'NotFoundLayout')}>
        <p><Link to={ ROUTE_HOME }><Translation name="Back" ns="NotFoundLayout" /></Link></p>
    </InfoMsg>
  </section>
);


export default NotFoundLayout;
