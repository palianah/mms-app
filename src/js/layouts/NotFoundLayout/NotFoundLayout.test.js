import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'
import NotFoundLayout from './NotFoundLayout';
import '../../components/Translation/testData';

configure({ adapter: new Adapter() });


describe('<NotFoundLayout />', () => {
  test('Renders without crashing', () => {
    const wrapper = shallow(<NotFoundLayout />);
    expect(wrapper).toHaveLength(1);
  });

  test('Matches snapshot', () => {
    expect(renderer.create(<MemoryRouter><NotFoundLayout /></MemoryRouter>).toJSON()).toMatchSnapshot();
  });
});
