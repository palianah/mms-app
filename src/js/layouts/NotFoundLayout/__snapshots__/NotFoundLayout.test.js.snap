// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundLayout /> Matches snapshot 1`] = `
<section
  className="NotFoundLayout"
>
  <section
    className="InfoMsg"
  >
    <i
      className="Icon icofont icofont-ui-block"
    />
    <div
      className="InfoMsg__msg"
    >
      ?Message:NotFoundLayout
    </div>
    <div
      className="InfoMsg__sub"
    >
      <p>
        <a
          href="/"
          onClick={[Function]}
        >
          ?Back:NotFoundLayout
        </a>
      </p>
    </div>
  </section>
</section>
`;
